### ISSUE TITLE
watts is not OS independent

### ISSUE THREAD

**Issue by samgdotson** [watts-dev/watts#51]
Currently, the `setup.cfg` file specifies `watts` as "OS Independent." However, there are at least two issues preventing Windows from supporting `watts`:

1. The `select.select([p.stdout, p.stderr])` call breaks because select.select [cannot work with streams](https://stackoverflow.com/questions/22251809/python-select-select-on-windows) on windows. The workaround for this is a [non-blocking pipe](https://stackoverflow.com/questions/16285362/problems-with-select-select-on-windows) which is addressed by @paulromano branch `nonblocking-pipe` (which was originally motivated by #49).
2. The `fcntl`  is not supported by windows. There may be some [substitutes](https://stackoverflow.com/questions/45228395/error-no-module-named-fcntl). This appears integral to @paulromano's aforementioned non-blocking pipe.

This issue can be closed when the developers decide to either
- [ ] add support for windows machines 
- [ ] indicate which operating systems are required

**Comment by nstauff** [comment]
@paulromano - can you check this issue and maybe discuss with Sam about potential solutions.

**Comment by paulromano** [comment]
Yes, I'm working on a resolution for this and plan to put in a PR
